dotnet3.5/S3/CopyObject/CopyObject.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+// snippet-start:[S3.dotnet35.CopyObject]
+using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+using System;
+using System.Threading.Tasks;
+
+namespace CopyObject
+{
+    public class CopyObject
+    {
+        // Specify your bucket region (an example region is shown).
+        private static readonly RegionEndpoint BUCKET_REGION = RegionEndpoint.USWest2;
+        private static IAmazonS3 _s3Client;
+
+        // Specify the names for the Simple Storage Service objects
+        private const string SOURCE_BUCKET_NAME = "*** provide the name of the bucket with source object ***";
+        private const string DESTINATION_BUCKET_NAME = "*** provide the name of the bucket to copy the object to ***";
+        private const string SOURCE_OBJ_KEY = "*** provide the name of object to copy ***";
+        private const string DESTINATION_OBJ_KEY = "*** provide the destination object key name ***";
+
+        static void Main()
+        {
+            _s3Client = new AmazonS3Client(BUCKET_REGION);
+
+            Console.WriteLine($"Copying {SOURCE_OBJ_KEY} from {SOURCE_BUCKET_NAME} to ");
+            Console.WriteLine($"{DESTINATION_BUCKET_NAME} as {DESTINATION_OBJ_KEY}");
+
+            var response = CopyingObjectAsync(
+                _s3Client,
+                SOURCE_OBJ_KEY,
+                DESTINATION_OBJ_KEY,
+                SOURCE_BUCKET_NAME,
+                DESTINATION_BUCKET_NAME);
+
+            Console.WriteLine("Copy complete!");
+
+        }
+
+        public static async Task<CopyObjectResponse> CopyingObjectAsync
```

Comment:
For "modular" functions such as this one, I encourage preceding it with a comment block describing what the function's purpose is, what each of any of the input parameters represent, what the return value if any is and what it represents. This is especially helpful for folks who skip the "Main" calling function and quickly cut and paste this function into their own code. If you do provide such comment blocks, be sure to format them in a way that's typical for the programming language (like you might need to preface input parameters with special symbols, etc.). Not sure what it is for .NET/C# these days though.

Reply:
Using documentation comments and adding an example of the call to show what the param values should look like.